components/encoder/single/single_encoder.go

```diff
@@ -152,6 +153,7 @@ func (e *Encoder) Reconfigure(
 	if err != nil {
 		return err
 	}
+	e.board = board
```

Comment:
Same suggestion in this file: leave the board as a local variable, and don't store it in the struct. Local variables are easier to reason about because they can't have race conditions from other things interacting with the struct.